lib/restforce/concerns/api.rb

```diff
@@ -361,12 +361,17 @@ def upsert(*args)
       # error returned if the external ID provided matches multiple records (in which
       # case the conflicting IDs can be found by looking at the response on the error)
       def upsert!(sobject, field, attrs)
-        external_id = attrs.
-          fetch(attrs.keys.find { |k, v| k.to_s.downcase == field.to_s.downcase }, nil)
-        attrs_without_field = attrs.
-          reject { |k, v| k.to_s.downcase == field.to_s.downcase }
-        response = api_patch "sobjects/#{sobject}/#{field}/#{URI.encode(external_id)}",
-                             attrs_without_field
+        external_id = attrs.delete(field.to_sym)
```

Comment:
Could we otherwise not document that you need to provide symbol keys and a symbol field name? That feels a bit nicer than having to do these comparisons every time.